mailpile/mailutils.py

```diff
@@ -1347,6 +1347,18 @@ def get_message_tree(self, want=None, tree=None):
 
                 if mimetype == 'text/html':
                     if want is None or 'html_parts' in want:
+                        # Find and delete possibly conflicting xml encoding
+                        # declaration to prevent lxml ValueError.
+                        # e.g. <?xml version="1.0" encoding="ISO-8859-1"?>
+                        re_encoding=re.compile('<\?xml version=[^ ?>]*'+
+                                                '((?! +encoding=) [^ ?>]*)*'+
+                                                '( +encoding=[^ ?>]*)')
+                        enc_match=re_encoding.match(payload)
```

Comment:
Re the above two comments - I will do some work on this - thanks for the suggestions.